Bottom: 490158a7208d10d874df48ccc2fbf1de8d399913
Top:    80d2b2ad221180dcd6e6aba0f482246576de4d34
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-07-25 16:36:39 +0800

Refresh of tun-xdp-support

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 1e552fd..1645f42 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1456,6 +1456,8 @@ static ssize_t tun_get_user(struct tun_struct *tun, struct tun_file *tfile,
 			this_cpu_inc(tun->pcpu_stats->rx_dropped);
 			return PTR_ERR(skb);
 		}
+		if (!skb)
+			return total_len;
 	} else {
 		if (!zerocopy) {
 			copylen = len;
